### DISCUSSION TITLE
Feature request: support syntax highlight for go:embed

### DISCUSSION THREAD

**Discussion by maito1201** [golang/vscode-go#1156] [Discussions]
I think there may be a better place to make this suggestion.
If you have a good idea, please let me know.

## background

`go:embed` is released at go 1.16🎉
https://golangtutorial.dev/tips/embed-files-in-go/

I think it is Difficult to distinguish between `go:embed` and comment, especially if `go:embed` mistypes.

It works as `go:embed`.
`//go:embed something`

It is just a comment.
`// go:embed something`

## Feature request

I want new syntax highlight for `go:embed`.

For example, When the input content is evaluated as `go:embed`, the color of highlight will change from green(comment color).

I have no strong opinion about color preferences,
But I think it would be nice if it changed to the same color as reserved word or something.

**Comment by hyangah** [comment]
Currently syntax highlighting is not handled by this extension, but I think `gopls` can provide a better syntax highlighting near future utilizing the semantic token. @pjweinb @stamblerre What do you think about this feature request?

Another direction I was thinking about was to provide a document link - so users can easily access the source file/directory of the embedded data.

**Reply by pjweinb** [reply]
it's plausible to have some special attention for //go:embed (and all the other comments that are sensitive to whitespace). Maybe gopls should warn about an unexpected space.

**Reply by maito1201** [reply]
Thank you for replying!
Certainly go repository seems to develop something  to make notification for go:embed like it.
https://github.com/golang/go/issues/43980
If I couldn’t find issue about syntax highlight in gopls github, I will make conversation at gopls .
